src/renderers/dom/fiber/ReactDOMFiber.js

```diff
@@ -96,6 +96,20 @@ function validateContainer(container) {
   }
 }
 
+function shouldAutoFocusHostComponent(
+  type : string,
+  props : Props,
+) : boolean {
+  switch (type) {
+    case 'button':
+    case 'input':
+    case 'select':
+    case 'textarea':
+      return !!(props : any).autoFocus;
```

Comment:
We should expand the `Props` type to include this as optional instead of relying on `any` here.